app/code/Magento/CatalogUrlRewrite/Observer/ProductProcessUrlRewriteSavingObserver.php

```diff
@@ -32,20 +40,38 @@ class ProductProcessUrlRewriteSavingObserver implements ObserverInterface
      */
     private $productUrlPathGenerator;
 
+    /**
+     * @var StoreManagerInterface $storeManager
+     */
+    private $storeManager;
+
+    /**
+     * @var StoreWebsiteRelationInterface
+     */
+    private $storeWebsiteRelation;
+
     /**
      * @param ProductUrlRewriteGenerator $productUrlRewriteGenerator
      * @param UrlPersistInterface $urlPersist
      * @param ProductUrlPathGenerator|null $productUrlPathGenerator
+     * @param StoreManagerInterface|null $storeManager
+     * @param StoreWebsiteRelationInterface|null $storeWebsiteRelation
      */
     public function __construct(
         ProductUrlRewriteGenerator $productUrlRewriteGenerator,
         UrlPersistInterface $urlPersist,
-        ProductUrlPathGenerator $productUrlPathGenerator = null
+        ProductUrlPathGenerator $productUrlPathGenerator = null,
```

Comment:
This class isn’t marked with `@api` tag, could you make all constructor arguments not optional?